src/AutoloadGenerator.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Flex;
+
+use Composer\Autoload\AutoloadGenerator as BaseAutoloadGenerator;
+use Composer\Config;
+use Composer\Installer\InstallationManager;
+use Composer\Package\PackageInterface;
+use Composer\Repository\InstalledRepositoryInterface;
+
+/**
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+class AutoloadGenerator extends BaseAutoloadGenerator
+{
+    private $generator;
+    private $platform;
+
+    public function __construct(parent $generator)
+    {
+        $this->generator = $generator;
+    }
+
+    public function setPlatform(string $platform)
+    {
+        $this->platform = $platform;
+    }
+
+    public function setDevMode($devMode = true)
+    {
+        $this->generator->setDevMode($devMode);
+    }
+
+    public function setClassMapAuthoritative($classMapAuthoritative)
+    {
+        $this->generator->setClassMapAuthoritative($classMapAuthoritative);
+    }
+
+    public function setApcu($apcu)
+    {
+        $this->generator->setApcu($apcu);
+    }
+
+    public function setRunScripts($runScripts = true)
+    {
+        $this->generator->setRunScripts($runScripts);
+    }
+
+    public function dump(Config $config, InstalledRepositoryInterface $localRepo, PackageInterface $mainPackage, InstallationManager $installationManager, $targetDir, $scanPsr0Packages = false, $suffix = '')
+    {
+        $this->generator->dump($config, $localRepo, $mainPackage, $installationManager, $targetDir, $scanPsr0Packages, $suffix);
+
+        if (null === $this->platform) {
+            return;
+        }
+
+        $autoloadFile = $config->get('vendor-dir').'/autoload.php';
+
+        $code = file_get_contents($autoloadFile);
+        $code = substr($code, \strlen("<?php\n"));
+
+        $platform = explode('.', $this->platform.'.0.0');
+        $platform = sprintf('%d%02d%02d', ...$platform);
+
+        file_put_contents($autoloadFile, <<<EOPHP
+<?php
+
+if (\PHP_VERSION_ID < $platform) {
+    echo sprintf("Fatal Error: PHP version $this->platform or higher is required, version %d.%d.%d is in use.\\n", PHP_MAJOR_VERSION, PHP_MINOR_VERSION, PHP_RELEASE_VERSION);
```

Comment:
I think the error message can be tweaked a bit to explain where this requirement comes from. I also think that "you are running version" explain it better than "version in use".
```suggestion
    echo sprintf("Fatal Error: PHP version $this->platform (or higher) is required in composer.json but you are running version %d.%d.%d.\\n", PHP_MAJOR_VERSION, PHP_MINOR_VERSION, PHP_RELEASE_VERSION);
```

Reply:
updated